pkg/sources/docker/docker.go

```diff
@@ -181,6 +198,35 @@ func (s *Source) processImage(ctx context.Context, image string) (imageInfo, err
 	return imgInfo, nil
 }
 
+// processHistoryEntry processes a history entry from the image configuration metadata.
+func (s *Source) processHistoryEntry(ctx context.Context, entryIndex int, entry v1.History, imgInfo imageInfo, chunksChan chan *sources.Chunk) error {
+	chunk := &sources.Chunk{
+		SourceType: s.Type(),
+		SourceName: s.name,
+		SourceID:   s.SourceID(),
+		SourceMetadata: &source_metadatapb.MetaData{
+			Data: &source_metadatapb.MetaData_Docker{
+				Docker: &source_metadatapb.Docker{
+					File:  fmt.Sprintf("image://config-file/history/%d", entryIndex),
+					Image: imgInfo.base,
+					Tag:   imgInfo.tag,
+					Layer: "",
+				},
+			},
+		},
+		Verify: s.verify,
+		Data:   []byte(entry.CreatedBy),
+	}
+
+	ctx.Logger().V(2).Info("scanning image history entry", "index", entryIndex)
+
+	if err := common.CancellableWrite(ctx, chunksChan, chunk); err != nil {
```

Comment:
Suggestion: Since we aren't including additional context with the returned error, we can skip the `err != nil` check and directly `return common.CancellableWrite(ctx, chunksChan, chunk)`.

Reply:
Thanks, will do. I often get caught in between leaving this error handling idiom and just returning directly, it's good to get a push to pick one :D